test/function.localstack.test.ts

```diff
@@ -0,0 +1,533 @@
+      }),
+    "value: hi",
+    { val: "hi" }
+  );
+
+  testFunctionResource(
+    "Call Lambda throw error",
+    (parent) =>
+      new Function(parent, "function", async () => {
+        throw Error("AHHHHHHHHH");
+      }),
+    { errorMessage: "AHHHHHHHHH", errorType: "Error" }
+  );
+
+  testFunctionResource(
+    "Call Lambda return arns",
+    (parent) => {
+      return new Function(parent, "function", async (_, context) => {
+        return context.functionName;
+      });
+    },
+    (context) => context.function
+  );
+
+  testFunctionResource(
+    "Call Lambda return arns",
+    (parent) => {
+      const bus = new EventBus(parent, "bus");
+      const busbus = new aws_events.EventBus(parent, "busbus");
+      const func = new Function(parent, "function", async () => {
+        return `${bus.eventBusArn} ${busbus.eventBusArn}`;
+      });
+
+      return {
+        func,
+        outputs: {
+          bus: bus.eventBusArn,
+          busbus: busbus.eventBusArn,
+        },
+      };
+    },
+    (context) => `${context.bus} ${context.busbus}`
+  );
+
+  testFunctionResource(
+    "templated tokens",
+    (parent) => {
+      const token = Token.asString("hello");
+      return new Function(parent, "function", async () => {
+        return `${token} stuff`;
+      });
+    },
+    "hello stuff"
+  );
+
+  testFunctionResource(
+    "numeric tokens",
+    (parent) => {
+      const token = Token.asNumber(1);
+      return new Function(parent, "function", async () => {
+        return token;
+      });
+    },
+    1
+  );
+
+  testFunctionResource(
+    "Call Lambda put events",
+    (parent) => {
+      const bus = new EventBus(parent, "bus");
+      return new Function(
+        parent,
+        "function",
+        async () => {
+          bus({
+            "detail-type": "detail",
+            source: "lambda",
+            detail: {},
+          });
+        },
+        localstackClientConfig
+      );
+    },
+    null
+  );
+
+  testFunctionResource(
+    "Call Lambda AWS SDK put event to bus with reference",
+    (parent) => {
+      const bus = new EventBus<any>(parent, "bus");
+
+      // Necessary to keep the bundle small and stop the test from failing.
+      // See https://github.com/sam-goodwin/functionless/pull/122
+      const putEvents = $AWS.EventBridge.putEvents;
+      const func = new Function(
+        parent,
+        "function",
+        async () => {
+          const result = putEvents({
+            Entries: [
+              {
+                EventBusName: bus.eventBusArn,
+                Source: "MyEvent",
+                DetailType: "DetailType",
+                Detail: "{}",
+              },
+            ],
+          });
+          return result.FailedEntryCount;
+        },
+        localstackClientConfig
+      );
+
+      bus.bus.grantPutEventsTo(func.resource);
+
+      return func;
+    },
+    0
+  );
+
+  // See https://github.com/sam-goodwin/functionless/pull/122
```

Comment:
fix link

Reply:
Fixed all of the previous links in #175.

Fixed all of the links from this change in the next commit.